src/modules/ChannelList/dux/reducers.js

```diff
@@ -170,16 +170,17 @@ export default function reducer(state, action) {
           allChannels: state.allChannels.filter(({ url }) => url !== channel?.url),
         };
       }
-      // if its only an unread message count change, dont push to top
-      if (unreadMessageCount === 0) {
-        const currentChannel = allChannels.find(({ url }) => url === channel?.url);
-        const currentUnreadCount = currentChannel && currentChannel.unreadMessageCount;
-        if (currentUnreadCount === 0) {
-          return {
-            ...state,
-            allChannels: state.allChannels.map((ch) => (ch.url === channel?.url ? channel : ch)),
-          };
-        }
+
+      if (
+        unreadMessageCount === 0
+        // Do not move to the top when marking as read the channel
+        && channel?.lastMessage?.sender?.userId !== state.currentUserId
+        // Move to the top when the current user but different peer sends the message
```

Comment:
Is there another logic supposed to be here below this comment line?

Reply:
That logic wasn't necessary. The currentChannel equals the channel in the allChannels